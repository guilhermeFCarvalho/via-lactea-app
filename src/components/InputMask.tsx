import { Input } from "native-base";
import React from "react";
import { TextInputProps } from "react-native";

import { mascaraData, mascaraPeso } from "../utils/Mascaras";

interface InputProps extends TextInputProps {
  mask: "data" | "peso",
  inputMaskChange: any;
}

const InputMask: React.FC<InputProps> = ({ mask, inputMaskChange, ... rest }) => {
  
  function pegarAlteracao(texto: string) {
    if(mask === 'data') {
      const textoFormatado = mascaraData(texto);
      inputMaskChange(textoFormatado)
    } if (mask === 'peso') {
      const textoFormatado = mascaraPeso(texto)
      inputMaskChange(textoFormatado)
    }
  }

  return (
    <Input
      onChangeText={text => pegarAlteracao(text)}
        { ...rest }
    >
    </Input>
  )
}

export default InputMask;