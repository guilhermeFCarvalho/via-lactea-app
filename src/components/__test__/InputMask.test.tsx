import React from 'react';
import { NativeBaseProvider } from 'native-base';
import renderer from 'react-test-renderer';
import InputMask from '../InputMask';

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: jest.fn(),
  };
});

test('renders correctly', () => {
  const inputMask = renderer
    .create(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <InputMask 
          mask="number"
          inputMaskChange={() => {}}>
        </InputMask>
      </NativeBaseProvider>,
    )
    .toJSON();
  expect(inputMask).toMatchSnapshot();
});