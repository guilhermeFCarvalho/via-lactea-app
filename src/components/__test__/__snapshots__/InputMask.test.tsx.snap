// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    isFocused={false}
    style={
      {
        "alignItems": "center",
        "borderColor": "#d4d4d4",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "overflow": "hidden",
      }
    }
  >
    <TextInput
      accessible={true}
      dataSet={{}}
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      placeholderTextColor="#a3a3a3"
      secureTextEntry={false}
      selectionColor="#1f2937"
      style={
        {
          "backgroundColor": "transparent",
          "color": "#171717",
          "flex": 1,
          "fontFamily": undefined,
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "400",
          "paddingBottom": 8,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 8,
          "width": "100%",
        }
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
